core/src/main/java/com/github/mizool/core/rest/errorhandling/ErrorHandler.java

```diff
@@ -103,49 +109,33 @@ private ErrorResponse handleWhitelistedException(Throwable t, WhiteListEntry whi
         Map<String, String> parameters = null;
         if (whiteListEntry.getShouldIncludeDetails())
         {
-            parameters = createExceptionParameters(t);
+            parameters = errorMapper.createExceptionParameters(t);
         }
         ErrorDto error = new ErrorDto(t.getClass().getName(), parameters);
-        ErrorMessageDto errorMessage = createErrorMessageDto(error);
+        ErrorMessageDto errorMessage = errorMapper.createErrorMessageDto(error);
         return new ErrorResponse(whiteListEntry.getStatusCode(), errorMessage);
     }
 
-    private ErrorResponse handleConstraintValidationError(ConstraintViolationException e)
+    private ErrorResponse handleConstraintViolationError(ConstraintViolationException e)
     {
         log.debug("Validation error", e);
-        ListMultimap<String, ErrorDto> errors = ArrayListMultimap.create();
-        for (ConstraintViolation<?> violation : e.getConstraintViolations())
-        {
-            recordConstraintViolation(violation, errors);
-        }
-        ErrorMessageDto errorMessage = createErrorMessageDto(errors);
+        ErrorMessageDto errorMessage = constraintViolationMapper.fromPojo(e.getConstraintViolations());
         return new ErrorResponse(SC_UNPROCESSABLE_ENTITY, errorMessage);
     }
 
-    private void recordConstraintViolation(ConstraintViolation<?> violation, ListMultimap<String, ErrorDto> target)
-    {
-        String errorId = violation.getConstraintDescriptor().getAnnotation().annotationType().getName();
-
-        Path.Node lastProperty = Iterators.getLast(violation.getPropertyPath().iterator());
-        String propertyName = lastProperty.getName();
-
-        ErrorDto errorDto = new ErrorDto(errorId, null);
-        target.put(propertyName, errorDto);
-    }
-
     private ErrorResponse handleRuleViolationError(RuleViolationException e)
     {
         log.debug("Rule violation error", e);
-        ListMultimap<String, ErrorDto> errors = ArrayListMultimap.create();
+        SetMultimap<String, ErrorDto> errors = HashMultimap.create();
```

Comment:
refactor this so that it uses the same structure as its cousin, `handleConstraintViolationError()`, including the separate mapper class